features/step_definitions/operators.rb

```diff
@@ -160,3 +160,34 @@
   cb_name = 'operator_channel_name' unless cb_name
   cb[cb_name] = cluster_version('version').channel.split('-')[1]
 end
+
+Given /^I wait(?: up to (\d+) seconds)? for operator "([^"]*)" to become conditions: "([^"]*)"$/ do | time, operator, conditions |
+  ensure_admin_tagged
+
+  expected = {}
+  # Parse the conditions to Hash value {Available"=>"True", "Progressing"=>"False", "Degraded"=>"False", "Upgradeable"=>"True"}
+  arr_conditions = conditions.split(/\s+/)    
+  raise "#### Invalid number of conditions!!! (Valid number:1~4)" unless arr_conditions.length <=4
+  if arr_conditions.any? { |x| x.include? "=" }
+    arr_conditions.map{|ar| ar.split('=')}.each {|k, v| expected[k]=v}
+  else
+    # Parse the conditions with "True False ..." input 
+    ["Available", "Progressing", "Degraded", "Upgradeable"][0,arr_conditions.length].zip(arr_conditions).map {|k, v| expected[k]=v}
+  end
+  expected.each {|k,v| raise "#### Please make sure the input status of condition is True or False!" unless ["True","False"].include? v }
+  raise "#### Without valid conditions, unable to continue to check !!!" unless expected.length >0
+
+  interval_time = 5 
+  time = Integer(time) rescue 60
+  interval_time = 20 if time > 100
+  actual_results = {}
+  success = wait_for(time, interval: interval_time) {
+    expected.keys.each do |c|
+        actual_results[c] = cluster_operator(operator).condition(type: c, cached: false)['status']
+    end
+    expected == actual_results
+  }
+  logger.info("#### Operator #{operator} Expected conditions: #{expected}")
+  logger.info("#### Operator #{operator} Actual conditions: #{actual_results}")
```

Comment:
Not only use logger.info from Pravin's PR https://github.com/openshift/verification-tests/pull/1648 , but also use stats[:seconds] and stats[:iterations], so that we easily know how much time the wait_for actually takes when success is true.

Reply:
Line 184 ~ 192, change to below:
```
  result = {
    props: {stats: {}}
  }         
  success = wait_for(time, interval: interval_time, stats: result[:props][:stats]) {
    ...
  }     
  stats = result[:props][:stats]
  logger.info "#### after #{stats[:seconds]} seconds and #{stats[:iterations]} iterations " <<
    "operator #{operator} is: #{actual_results}"
    raise "The #{operator} operator still didn't become #{expected} after #{time} seconds" unless success
```

Reply:
My test result:
```
$ cat features/myquicktest.feature
Feature: Test feature
  @admin
  Scenario: test kewang verification-tests/pull/1719
    Given I wait up to 20 seconds for operator "kube-apiserver" to become conditions: "Progressing=False"
    Given I wait up to 10 seconds for operator "kube-apiserver" to become conditions: "Progressing=True"
```
Got:
```
[08:25:50] INFO> #### after 17.005530637223274 seconds and 1 iterations operator kube-apiserver is: {"Progressing"=>"False"}
The kube-apiserver operator still didn't become {"Progressing"=>"True"} after 10 seconds (RuntimeError)
/data/src/github.com/openshift/verification-tests/features/step_definitions/mytest.rb:64:in `/^I wait(?: up to (\d+) seconds)? for operator "([^"]*)" to become conditions: "([^"]*)"$/' 
features/myquicktest.feature:5:in `Given I wait up to 10 seconds for operator "kube-apiserver" to become conditions: "Progressing=True"'
```

Reply:
Useful output.